x/twap/api_test.go

```diff
@@ -182,9 +236,106 @@ func (s *TestSuite) TestGetArithmeticTwap() {
 				s.twapkeeper.StoreNewRecord(s.Ctx, record)
 			}
 			s.Ctx = s.Ctx.WithBlockTime(test.ctxTime)
-			twap, err := s.twapkeeper.GetArithmeticTwap(s.Ctx, test.input.poolId,
+
+			var twap sdk.Dec
+			var err error
+
+			twap, err = s.twapkeeper.GetArithmeticTwap(s.Ctx, test.input.poolId,
 				test.input.quoteAssetDenom, test.input.baseAssetDenom,
 				test.input.startTime, test.input.endTime)
+
+			if test.expErrorStr != "" {
+				s.Require().Error(err)
+				s.Require().Contains(err.Error(), test.expErrorStr)
+				return
+			}
+			s.Require().NoError(err)
+			s.Require().Equal(test.expTwap, twap)
+		})
+	}
+}
+
+// TestGetArithmeticTwap tests if we get the expected twap value from `GetArithmeticTwap`.
+// We test the method directly by updating the accumulator and storing the twap records
+// manually in this test.
+// This test also includes testing the method `TestGetArithmeticTwapToNow`.
+func (s *TestSuite) TestGetArithmeticTwapToNow() {
+	type getTwapInput struct {
+		poolId          uint64
+		quoteAssetDenom string
+		baseAssetDenom  string
+		startTime       time.Time
+		endTime         time.Time
+	}
+
+	makeSimpleTwapToNowInput := func(startTime time.Time, isQuoteTokenA bool) getTwapInput {
+		quoteAssetDenom, baseAssetDenom := denom0, denom1
+		if isQuoteTokenA {
+			baseAssetDenom, quoteAssetDenom = quoteAssetDenom, baseAssetDenom
+		}
+		return getTwapInput{1, quoteAssetDenom, baseAssetDenom, startTime, startTime}
+	}
+
+	quoteAssetA := true
+
+	tests := map[string]struct {
+		recordsToSet []types.TwapRecord
+		ctxTime      time.Time
+		input        getTwapInput
+		expTwap      sdk.Dec
+		expErrorStr  string
+	}{
+		"(1 record) start time = record time": {
+			recordsToSet: []types.TwapRecord{baseRecord},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapToNowInput(baseTime, quoteAssetA),
+			expTwap:      sdk.NewDec(10),
+		},
+		"(1 record) to_now: start time > record time": {
+			recordsToSet: []types.TwapRecord{baseRecord},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapToNowInput(baseTime.Add(10*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(10),
+		},
+		"(2 record) to now: start time = second record time": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapToNowInput(baseTime.Add(10*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(5), // .1 for 10s, .2 for 10s
+		},
+		"(2 record) first record time < start time < second record time": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record},
+			ctxTime:      baseTime.Add(20 * time.Second),
+			input:        makeSimpleTwapToNowInput(baseTime.Add(5*time.Second), quoteAssetA),
+			// 10 for 5s, 5 for 10s = 100/15 = 6 + 2/3 = 6.66666666
+			expTwap: ThreePlusOneThird.MulInt64(2),
```

Comment:
Does it make sense to have a 3 record to now test?

Reply:
same call for this!